pad_api_data/pad_etl/processor/skill_info.py

```diff
@@ -648,8 +650,26 @@ def f(x):
         _, c = convert_with_defaults('auto_heal',
                                      arguments,
                                      auto_heal_backups)(x)
-        c['skill_text'] += fmt_duration(c['duration']) + 'recover ' + \
-            fmt_mult(c['percentage_max_hp'] * 100) + '% of max HP'
+        skill_text = ''
+        if c['duration']:
+            skill_text += fmt_duration(c['duration']) + 'recover ' + \
+                fmt_mult(c['percentage_max_hp'] * 100) + '% of max HP'
+        
+        if c['unbind'] and c['awoken_unbind'] and skill_text:
+            skill_text += '; Reduce binds and awoken skill binds by {} turns'.format(c['unbind'])
+        elif c['unbind'] and c['awoken_unbind']:
+            skill_text += 'Reduce binds and awoken skill binds by {} turns'.format(c['unbind'])
+        elif c['unbind'] and skill_text:
+            skill_text += '; Reduce binds by {} turns'.format(c['unbind'])
+        elif c['unbind']:
+            skill_text += 'Reduce binds by {} turns'.format(c['unbind'])
+        elif c['awoken_unbind'] and skill_text:
+            skill_text += '; Reduce awoken skill binds by {} turns'.format(c['awoken_unbind'])
+        elif c['awoken_unbind']:
+            skill_text += 'Reduce awoken skill binds by {} turns'.format(c['awoken_unbind'])
+
+        c['skill_text'] += skill_text

```

Comment:
Please refactor this code a bit. At the start you can stick an 'if skill_text: skill_text += '; ' which will halve the number of cases.

You can also extract c['unbind'] and c['awoken_unbind'] to local variables to improve readability.